chainer/functions/concat.py

```diff
@@ -22,6 +23,38 @@ class Concat(function.Function):
     def __init__(self, axis=1):
         self.axis = axis
 
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() > 0)
+        type_check.expect(in_types[0].ndim >
+                          type_check.IntVariable(self.axis, 'axis'))
```

Comment:
Can I write `in_types[0].ndim.eval > self.axis`? I think it is better because value of left hand side is used afterwards.